### PR TITLE
perf(ci): Build snuba image only once in CI

### PR DESCRIPTION
**Pull request by john-z-yang**
### Overview

The snuba GHA that runs tests seems to be waiting on a GHA that builds a snuba image. But for some reason those tests also rebuilds the image again. This PR makes some changes to the test jobs definition to reuse the image that previous steps create.
The PR also includes some commits where I upload bad tests to make sure the CI is indeed using the right image.

This appears to have lowered the overall CI time from ~31 minutes to ~22 minutes.
When re-running tests jobs due to flakes, each test jobs should be at least 8 minutes quicker.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -199,24 +199,18 @@ jobs:
...
-          tags: snuba-test
-          load: true
-          target: testing
-          cache-from: |
-            type=registry,ref=ghcr.io/getsentry/snuba-ci:${{ github.sha }}
-            type=registry,ref=ghcr.io/getsentry/snuba-ci:${{ needs.snuba-image.outputs.branch }}
-            type=registry,ref=ghcr.io/getsentry/snuba-ci:latest
+      - name: Pull the image
+        env:
+          IMAGE_URL: us-central1-docker.pkg.dev/sentryio/snuba/image:${{ github.event.pull_request.head.sha || github.sha }}
```
Comment (hubertdeng123) [inline .github/workflows/ci.yml:208]: I think you may need to use `ghcr.io/getsentry/snuba-ci:${{ github.sha }}` instead, that seems to be where the image is being pushed in step `snuba-image`

**Comment by john-z-yang** [comment]
Failed (as hoped)
image

**Comment by john-z-yang** [comment]
Failed (as hoped)
image

File: .github/workflows/ci.yml (hunk 1 above)
Comment (john-z-yang) [reply .github/workflows/ci.yml]: Thanks, fixed!

**Review by volokluev — APPROVED** [review]
Well done

**Review by hubertdeng123 — APPROVED** [review]
looks good

**Pull request merged**